linux_os/guide/system/accounts/accounts-pam/password_quality/password_quality_pwquality/accounts_password_pam_retry/ansible/shared.yml

```diff
@@ -3,15 +3,38 @@
 # strategy = configure
 # complexity = low
 # disruption = medium
+{{% if 'ubuntu' in product %}}
+{{% set configuration_files = ["common-password"] %}}
+{{% elif product in ['ol8','ol9','rhel8', 'rhel9'] %}}
+{{% set configuration_files = ["password-auth","system-auth"] %}}
+{{% else %}}
+{{% set configuration_files = ["system-auth"] %}}
+{{% endif %}}
+
 {{{ ansible_instantiate_variables("var_password_pam_retry") }}}
 
-{{% if product in ['rhel8', 'rhel9'] -%}}
+{{% if product in ['ol8','ol9','rhel8', 'rhel9'] -%}}
 - name: Ensure PAM variable retry is set accordingly
   ansible.builtin.lineinfile:
     create: yes
     dest: "/etc/security/pwquality.conf"
-    regexp: '^#?\s*retry'
+    regexp: '^\s*retry'
     line: "retry = {{ var_password_pam_retry }}"
+{{% for cfile in configuration_files %}}
+{{{ ansible_remove_pam_module_option_configuration(pam_file='/etc/pam.d/' ~ cfile,
+                                                   group='password',
+                                                   control=".*",
+                                                   module='pam_pwquality.so',
+                                                   option='retry') }}}
+{{% endfor %}}
 {{%- else -%}}
-{{{ ansible_ensure_pam_module_option('/etc/pam.d/system-auth', 'password', 'requisite', 'pam_pwquality.so', 'retry', '{{ var_password_pam_retry }}', '^account.*required.*pam_permit.so') }}}
+{{% for cfile in configuration_files %}}
+{{{ ansible_ensure_pam_module_option('/etc/pam.d/' ~ cfile,
+                                     'password',
+                                     'requisite',
+                                     'pam_pwquality.so',
+                                     'retry',
+                                     '{{ var_password_pam_retry }}'
+                                     '^\s*account') }}}
```

Comment:
This value for `after_match` causes duplicated lines in RHEL, after remediation:

```
cat /etc/pam.d/system-auth
account     required      pam_unix.so
password     requisite    pam_pwquality.so
account     sufficient    pam_localuser.so
password     requisite    pam_pwquality.so
account     sufficient    pam_succeed_if.so uid < 1000 quiet
password     requisite    pam_pwquality.so
account     required      pam_permit.so
password     requisite    pam_pwquality.so
```
So, I would recommend to keep the regex as more specific as possible. The `'^account.*required.*pam_permit.so'` is valid for RHEL cases. Maybe a jinja2 conditional could be included if OL has a different content on respective files.

Reply:
It was strange to me as the doc said it only insert the line after the last match, I found out that I was missing the comma between `'{{ var_password_pam_retry }}'` and `'^\s*account'`. Now it works as expected

Reply:
I confirmed it is working fine now.